ui/imports/shared/stores/DAppsStore.qml

```diff
@@ -33,6 +33,27 @@ QObject {
         return controller.signTypedDataV4(address, password, typedDataJson)
     }
 
+    // Remove leading zeros from hex number as expected by status-go
+    function stripLeadingZeros(hexNumber) {
+        let fixed = hexNumber.replace(/^0x0*/, '0x')
+        return fixed == '0x' ? '0x0' : fixed;
+    }
+
+    // Returns the hex encoded signature of the transaction or empty string if error
+    function signTransaction(topic, id, address, chainId, password, txObj) {
+        // Strip leading zeros from numbers as expected by status-go
+        let tx = {
+            data: txObj.data,
+            from: txObj.from,
+            gasLimit: stripLeadingZeros(txObj.gasLimit),
+            gasPrice: stripLeadingZeros(txObj.gasPrice),
+            nonce: stripLeadingZeros(txObj.nonce),
+            to: txObj.to,
+            value: stripLeadingZeros(txObj.value)
+        }
+        return controller.signTransaction(address, chainId, password, JSON.stringify(tx))
```

Comment:
I was looking for how the connection to the service was made, and after analysis of the file, I realized this store doesn't follow the common pattern below. Essentially the connection to the module is omitted here ?

```mermaid
graph TD
    subgraph Business Logic Layer
        DataController["Data Controller<br>(communicates with service/s and keeps necessary data)"]
        ModuleLogic["Module<br>(main logic for this module)"]
        DataController --> ModuleLogic
        ModuleLogic --> DataController
    end

    subgraph Adapter Layer
        View
        Models
        View --> Models
    end

    subgraph Service Layer
       AccountService --> DataController
       ContactsService --> DataController
       ChatService --> DataController
       CommunityService --> DataController
       MessageService --> DataController
       AccountService --> DataController
    end

    subgraph Presentation Layer
       Qml --> View
    end

    ModuleLogic --> View
```

Reply:
There is no module for the current implementation. Just the controller providing access to the persistence API. I removed the module because it wasn't doing anything just duplicating methods and made me chase some init issues step that were not doing anything. We might introduce it when we feel we need it.

Note that unlike most of the other modules that have as main services consuming and generating data from/to status-go the dapps has the main source of data the QML based SDK interface so I chose to have also the logic in QML not to have unnecessary round trips between nim and QML